### PR TITLE
release: document major changes

### PR DESCRIPTION
**Pull request by nashif**
added highlights to the 3.4 release notes.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: doc/releases/release-notes-3.4.rst [hunk 1]
```diff
@@ -9,6 +9,31 @@ We are pleased to announce the release of Zephyr version 3.4.0.
 
 Major enhancements with this release include:
 
+* Input subsystem: handles input events from various types of input devices and
+  distribute them to other threads in the application.
+* Barrierr API: add architecture agnostic API for data memory barriers.
+* USB Device support:
+
+  * USB device controller API (UDC API)
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.4.rst:17]: nit: dot at the end of the sentences? Not that the rest of the file is coherent about it, but individual sections tend to be coherent within themselves a bit, plus this one stands out

**Comment by cfriedt** [comment]
Good catch @nashif !

**Review by cfriedt — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

File: doc/releases/release-notes-3.4.rst (hunk 1 above)
Comment (nashif) [reply doc/releases/release-notes-3.4.rst:17]: yeah, need to sanitize the whole file in one go.

File: doc/releases/release-notes-3.4.rst [hunk 2]
```diff
@@ -9,6 +9,31 @@ We are pleased to announce the release of Zephyr version 3.4.0.
 
 Major enhancements with this release include:
 
+* Input subsystem: handles input events from various types of input devices and
+  distribute them to other threads in the application.
+* Barrierr API: add architecture agnostic API for data memory barriers.
```
[the remark this answers is not in the document]
Comment (nashif) [reply doc/releases/release-notes-3.4.rst:14]: I am going through the doc in its totality right now and fixing such issues.

**Pull request merged**